tests/kernel/gen_isr_table/testcase.yaml

```
@@ -64,3 +64,19 @@ tests:
     filter: CONFIG_SOC_FAMILY_RISCV_PRIVILEGED
     extra_configs:
       - CONFIG_GEN_IRQ_VECTOR_TABLE=n
+  arch.interrupt.gen_isr_table.bit_shift_2nd_level:
+    platform_allow: qemu_riscv32
+    filter: CONFIG_SOC_FAMILY_RISCV_PRIVILEGED
+    extra_configs:
+      - CONFIG_1ST_LEVEL_INTERRUPT_BITS=10
+      - CONFIG_2ND_LEVEL_INTERRUPT_BITS=10
+  arch.interrupt.gen_isr_table.bit_shift_3rd_level:
+    platform_allow: qemu_riscv32
+    filter: CONFIG_SOC_FAMILY_RISCV_PRIVILEGED
+    extra_configs:
+      - CONFIG_MULTI_LEVEL_INTERRUPTS=y
+      - CONFIG_2ND_LEVEL_INTERRUPTS=y
+      - CONFIG_3RD_LEVEL_INTERRUPTS=y
+      - CONFIG_1ST_LEVEL_INTERRUPT_BITS=10
+      - CONFIG_2ND_LEVEL_INTERRUPT_BITS=10
+      - CONFIG_3RD_LEVEL_INTERRUPT_BITS=9
```

different numbers would allow better test of the formulas
```suggestion
      - CONFIG_1ST_LEVEL_INTERRUPT_BITS=11
      - CONFIG_2ND_LEVEL_INTERRUPT_BITS=10
      - CONFIG_3RD_LEVEL_INTERRUPT_BITS=9
```

---

I don't think this really makes any difference.